docs/api-reference.md

```diff
@@ -106,15 +108,49 @@ Roact.createRef() -> Ref
 
 Creates a new reference object that can be used with [Roact.Ref](#roactref).
 
+### Roact.setGlobalConfig
+```
+Roact.setGlobalConfig(configValues: Dictionary<string, bool>) -> void
+```
+
+The entry point for configuring Roact. Roact currently applies this to everything using this instance of Roact, so be careful using this with a project that has multiple consumers of Roact.
+
+Once config values are set, they will apply from then on. This is primarily useful when developing as it can enable features that validate your code more strictly. Most of the settings here incur a performance cost and should typically be disabled in production environments.
+
+The following are the valid config keys that can be used, and what they do.
+
+#### typeChecks
+
```

Comment:
We should prune these extra newlines to make it more clear where sections are when reading in plaintext.